sky/services/platform/ios/path_provider_impl.mm

```diff
@@ -0,0 +1,50 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "sky/services/platform/ios/path_provider_impl.h"
+#include "base/mac/scoped_nsautorelease_pool.h"
+
+#include <Foundation/Foundation.h>
+
+#include <string>
+
+namespace flutter {
+namespace platform {
+
+PathProviderImpl::PathProviderImpl(mojo::InterfaceRequest<PathProvider> request)
+    : binding_(this, request.Pass()) {}
+
+PathProviderImpl::~PathProviderImpl() {}
+
+void PathProviderFactory::Create(mojo::ApplicationConnection* connection,
+                                 mojo::InterfaceRequest<PathProvider> request) {
+  new PathProviderImpl(request.Pass());
+}
+
+static std::string GetDirectoryOfType(NSSearchPathDirectory dir) {
+  base::mac::ScopedNSAutoreleasePool pool;
```

Comment:
For calls where I am not sure if the objects returned are autoreleased, I brace the scope with an explicit autorelease pool. Thats because I don't make assumptions about the thread on which the service impl calls are made. Not all threads have a runloop that has an implicit autoreleasepool. The solution is to push an implicit pool when the runloop wakes up (the `kCFRunLoopAfterWaiting` equivalent)  and pop it when before the thread goes back to sleep. But we make ObjC calls so infrequently anyway.

In this case specifically, I am pretty sure the paths array is not autoreleased. So I can probably get away without using a pool. But its not documented, so I dont assume it.